src/d3/hapticplot.d3.ts

```diff
@@ -0,0 +1,137 @@
+// run `tsc Hapticplot.ts` to compile into Hapticplot.js file
+
+// for running on browser
+// import * as _d3 from 'd3';
+
+// declare global {
+// const d3: typeof _d3;
+// }
+
+// for running on unit tests
+import * as d3 from 'd3';
+import * as THREE from 'three';
+
+export class Hapticplot{
+    private data: number[];
+    private shape: string;
+    private container: HTMLElement | null;
+    private hscale: d3.ScaleLinear<number, number>;
+
+  constructor(shape: string) {
+    this.shape = shape;
+  }
+  init(container: HTMLElement | null, data: number[]){
+    this.data = data;
+    this.container = container;
+    // create a scale so that there is correspondence between data set and screen render
+    this.hscale = d3.scaleLinear();
+    this.hscale.domain([0, d3.max(this.data) as number])       // max of dataset
+    .range([0, 100]);                                      // linear mapping of data set values to values from 0 to 10
+    this.createSky();
+    this.generatePts();
+    this.setColor('blue');
+    this.setRadius(0.1);
+    this.setInteraction();
+    this.addEventListeners();
+    this.createGridPlane();
+  }
+
+  private generatePts() {
+     /*
+     Generates points based on initilization data
+      - "enter" identifies any DOM elements to be added when # array elements & # DOM elements don't match
+      - select all generated entities and updates their position based on ingested data
+    */
+    d3.select(this.container).selectAll(this.shape).data(this.data).enter().append(this.shape);
+    d3.select(this.container).selectAll(this.shape).attr('position', (d, i) => {
+      const x = i/10;
+      const y = this.data[i];
+      const z = -1;
+      return `${x} ${y} ${z}`;
+    });
+  }
+  private setColor(color) {
+    /*
+    Adds given color property to all generated entities
+    */
+    d3.select(this.container).selectAll(this.shape).attr('color', () => {
```

Comment:
Seems like you could make a utility method that does this selectAll thing, since you are doing it so much

Reply:
done! but I also condensed the functions that were using it in to one function that only does it once, so this might no longer be necessary. might be nice to leave in for readability though.